pandas/core/generic.py

```diff
@@ -5740,6 +5741,134 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def convert_dtypes(
+        self: FrameOrSeries,
+        infer_objects: bool_t = True,
+        convert_string: bool_t = True,
+        convert_integer: bool_t = True,
+        convert_boolean: bool_t = True,
+    ) -> FrameOrSeries:
+        """
+        Convert columns to best possible dtypes, optionally using dtypes supporting
+        ``pd.NA``.
+
```

Comment:
I would also add a note here in the docstring that the output of the function can change in the future when new nullable dtypes are added in the future (which I think we should then add to this function, to expand the inference automatically, with a new option to turn it off).

Reply:
done